### PR TITLE
Ability to select all media in selection mode

### PR DESCRIPTION
**Pull request by phraemer**
### First time contributor checklist

- [x] I have read the [README](https://github.com/signalapp/Signal-iOS/blob/main/README.md) and [CONTRIBUTING](https://github.com/signalapp/Signal-iOS/blob/main/CONTRIBUTING.md) documents
- [x] I have signed the [Contributor Licence Agreement](https://signal.org/cla/)

### Contributor checklist

- [x] My commits are rebased on the latest main branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * iPhone 13, iOS 17.0.3
 * Simulator iPhone 15, iOS 17.0

- - - - - - - - - -

### Description

When deleting media it can take an extraordinary amount of time to select all items to delete as there is currently no way to select all. This is not only inconvenient but impacts user security when it may be necessary to delete all media from a conversation quickly without losing the text of the conversation. It's also a very common UI pattern to be able to select all items.

This PR adds a text button to the task bar allowing the user in one tap to select all the items in the view. When any items are selected the text of the buttons allows the user to select none.
The button is placed on the left away from the Delete button to prevent any accidental presses.

https://github.com/signalapp/Signal-iOS/assets/143631/c40f9fcd-9fca-4b83-ad82-ce24c3b9b95d

### CODE REVIEW

File: Signal/src/ViewControllers/MediaGallery/MediaGalleryAccessoriesHelper.swift [hunk 1]
```diff
@@ -569,6 +573,33 @@ public class MediaGalleryAccessoriesHelper {
...
+        title: CommonStrings.selectAllButton,
+        style: .plain,
+        target: self,
+        action: #selector(didPressSelectAll)
+    )
+  
+    @objc
+    private func didPressSelectAll(_ sender: Any) {
+      Logger.debug("")
+      if viewController?.hasSelection == false {
```
Comment (pawisoon) [inline Signal/src/ViewControllers/MediaGallery/MediaGalleryAccessoriesHelper.swift:589]:
```suggestion
      if viewController?.hasSelection  {
          viewController?.selectNone()
      } else {
          viewController?.selectAll()
      }
```

**Review by pawisoon — APPROVED** [review]
See my code clean-up suggestion :) 
Approved 👍

**Comment by Excelbrains** [comment]
I hope that this welcome feature is added in short notice, since a lot of users are struggling with cleansing their media in Signal!

**Comment by iSack97** [comment]
Please merge this asap, a lot of people needs this

**Comment by phraemer** [comment]
Belated happy new year to the Signal team 😚🎉

How are we looking at getting this in?
It will help so many people!

**Comment by phraemer** [comment]
@harry-signal please please please can we get this merged? It's so frustratingly difficult and time consuming to manage storage space in Signal on iOS and this will be such a huge help for such a small change.

**Comment by tohrxyz** [comment]
+1 hoping for this to come asap

**Comment by phraemer** [comment]
I am absolutely begging you to get this in please!
It's utterly impractical to manually select items for what are now years of stuff in a chat.

@harry-signal ... anyone? Just let me know what I can do to move this simple but incredibly useful change forward.

Thanks.

**Comment by iSack97** [comment]
I know devs are working on other backup functionality, but this is a fucking joke. They just need to push this and a lot of people would have easier lifes

**Comment by phraemer** [comment]
Approaching a year now and no feedback. Doesn’t look like you have merged any contributions to iOS Signal for more than a year either. Some communication from anyone from Signal would be greatly appreciated.

**Comment by harry-signal** [comment]
This feature is coming in the first release of the new year! Thanks for the feedback folks! (Thanks for the timely reminder, stale bot!)

**Comment by phraemer** [comment]
I’m really happy this is coming but would really really **really** appreciate and recommend better communication. Please!

**Pull request closed without merging**